src/main/java/org/evgenysav/infrastructure/core/impl/ObjectFactoryImpl.java

```diff
@@ -16,15 +17,24 @@ public class ObjectFactoryImpl implements ObjectFactory {
 
     private final Context context;
     private final List<ObjectConfigurator> objectConfigurators = new ArrayList<>();
+    private final List<ProxyConfigurator> proxyConfigurators = new ArrayList<>();
 
     @SneakyThrows
     public ObjectFactoryImpl(Context context) {
         this.context = context;
         Scanner scanner = context.getConfig().getScanner();
-        Set<Class<? extends ObjectConfigurator>> subTypesOfConfigurators = scanner.getSubTypesOf(ObjectConfigurator.class);
-        for (var configurator : subTypesOfConfigurators){
+        Set<Class<? extends ObjectConfigurator>> subTypesOfConfigurators =
+                scanner.getSubTypesOf(ObjectConfigurator.class);
+        for (var configurator : subTypesOfConfigurators) {
             objectConfigurators.add(configurator.newInstance());
         }
+
+        Set<Class<? extends ProxyConfigurator>> subTypesOfProxyConfigurators =
+                scanner.getSubTypesOf(ProxyConfigurator.class);
+
+        for (Class<? extends ProxyConfigurator> implementation : subTypesOfProxyConfigurators) {
+            proxyConfigurators.add(implementation.newInstance());
+        }
     }
```

Comment:
Might be worth using the stream here it is worth using stream here
for example 
objectConfigurators.addAll( subTypesOfConfigurators.stream()
                                                                                      .map( configurator::newInstance )
                                                                                      .collect( Collectors.toSet() ) );
                                                     
proxyConfigurators.addAll( subTypesOfProxyConfigurators.stream()
                                                                                              .map( proxyConfigurator::newInstance )
                                                                                              .collect( Collectors.toSet() ) );

Reply:
also you can extract this logic in a separate Method public static void addConfigurators(Set<? extends T> subTypes) in some utils class and make wrappers 
public static void addObectConfigurators(Set<? extends ObjectConfigurator> subTypes){
addConfigurators(subTypes);
};
AND
public static void addProxytConfigurators(Set<? extends ProxyConfigurator> subTypes){
addConfigurators(subTypes);
};

And it worth make context of configurators (something like enum) with configurators needed to create and using stream to iterate and add all needed instances